### PR TITLE
Add documentation to several activation functions

### PR DESCRIPTION
**Pull request by vkk800**
Some of the activation function documentation was either missing or not very informative. This commit improves it a bit. Some of the activations I still left to minimal documentation because they are self-explanatory (like tanh activation).

None of the actual code was changed, only documentation.

### CODE REVIEW

File: keras/activations.py [hunk 1]
```diff
@@ -15,8 +15,8 @@ def softmax(x, axis=-1):
     """Softmax activation function.
 
     # Arguments
-        x : Tensor.
-        axis: Integer, axis along which the softmax normalization is applied.
+        - x : Input tensor.
+        - axis: Integer, axis along which the softmax normalization is applied.
```
Comment (fchollet) [inline keras/activations.py]: Don't structure arguments as a bullet-point list, each line should start with the argument name.

File: keras/activations.py [hunk 2]
```diff
@@ -36,17 +36,35 @@ def softmax(x, axis=-1):
 
 
 def elu(x, alpha=1.0):
+    """Exponential linear unit.
+
+    # Arguments
+        - x: Input tensor.
+        - alpha: A scalar, slope of negative section.
```
Comment (fchollet) [inline keras/activations.py]: Same comment

File: keras/activations.py [hunk 3]
```diff
@@ -36,17 +36,35 @@ def softmax(x, axis=-1):
...
+
+    # Returns
+        The exponential linear activation: `x` if `x > 0` and `alpha * (exp(x)-1)` if `x < 0`
```
Comment (fchollet) [inline keras/activations.py]: Make lines shorter than 80 char, here and throughout these changes.

File: keras/activations.py [hunk 4]
```diff
@@ -36,17 +36,35 @@ def softmax(x, axis=-1):
...
 
 
 def selu(x):
-    """Scaled Exponential Linear Unit. (Klambauer et al., 2017).
+    """Scaled Exponential Linear Unit (SELU).
+
+    SELU is equal to: `scale * elu(x, alpha)`, where alpha and scale are pre-defined constants.
+    The values of `alpha` and `scale` are chosen so that the mean and variance of the inputs are
+    preserved between two consecutive layers as long as the weights are initialized correctly
+    (see "lecun_normal" initialization) and the number of inputs is "large enough" (see
```
Comment (fchollet) [inline keras/activations.py]: Put `lecun_normal` inside backquotes `

File: keras/activations.py [hunk 5]
```diff
@@ -61,30 +79,74 @@ def selu(x):
...
     return K.softsign(x)
 
 
 def relu(x, alpha=0., max_value=None):
+    """(Leaky) Rectified Linear Unit.
+
+    # Arguments
+        - x: Input tensor
+        - alpha: Slope of the negative part. Defaults to zero.
+        - max_value: Maximum value for the output.
```
Comment (fchollet) [inline keras/activations.py]: Same comment about formatting

File: keras/activations.py [hunk 6]
```diff
@@ -61,30 +79,74 @@ def selu(x):
...
+        The softplus activation: `x / (abs(x) + 1)`
+    """
     return K.softsign(x)
 
 
 def relu(x, alpha=0., max_value=None):
+    """(Leaky) Rectified Linear Unit.
+
+    # Arguments
+        - x: Input tensor
```
Comment (fchollet) [inline keras/activations.py]: You normally end one-line descriptions with a period, so do it here too for consistency

File: keras/activations.py [hunk 7]
```diff
@@ -61,30 +79,74 @@ def selu(x):
...
+        - alpha: Slope of the negative part. Defaults to zero.
+        - max_value: Maximum value for the output.
+
+    # Returns
+        The (leaky) rectified linear unit activation: `x` if `x > 0`, `alpha * x` if `x < 0`.
+        If `max_value` is defined, the result is clamped to this value.
```
Comment (fchollet) [inline keras/activations.py]: clamped -> truncated

File: keras/activations.py [hunk 8]
```diff
@@ -61,30 +79,74 @@ def selu(x):
...
+    """Hard sigmoid activation function.
+
+    Faster to compute than sigmoid activation.
+
+    # Arguments
+        x: Input tensor
+
+    # Returns
+        Hard sigmoid activation: `0` if `x < -2.5`, `1` if `x > 2.5` and `0.2 * x + 0.5`
+        if `-2.5 <= x <= 2.5`.
```
Comment (fchollet) [inline keras/activations.py]: You can structure these different cases as a bullet point list

File: keras/activations.py (hunk 1 above)
Comment (vkk800) [reply keras/activations.py]: Not doing this results in formatting that looks bad. See for example the current documentation of the softmax function in https://keras.io/activations/ .
